main.go

```diff
@@ -105,29 +108,81 @@ func checkRegionFlag(meta *ec2Meta) {
 }
 
 func execute(ec2 *ec2Client, ecs *ecsClient) {
-	updateAndWrite(ec2, ecs)
-	var err error
-	err = runSignal()
+	if updateAndWrite(ec2, ecs) {
+		var err error
+		err = runSignal()
+		if err != nil {
+			log.Println("failed to run signal command")
+			log.Println("error: ", err)
+			switch err := err.(type) {
+			case *exec.ExitError:
+				log.Println(err.Stderr)
+			}
+			cleanupTempFile()
+			os.Exit(1)
+		}
+	}
+}
+
+func hasUpdate(containers []*container) bool {
+
+	containerHash, err := hashstructure.Hash(containers, nil)
+	if err != nil {
+		log.Println("unable to hash containers: ", err)
+	}
+
+	templateContents, err := ioutil.ReadFile(*templateFile)
+	if err != nil {
+		log.Println("unable to read template file: ", err)
+	}
+
+	templateHash, err := hashstructure.Hash(templateContents, nil)
 	if err != nil {
-		log.Println("failed to run signal command")
-		log.Println("error: ", err)
-		switch err := err.(type) {
-		case *exec.ExitError:
-			log.Println(err.Stderr)
+		log.Println("unable to hash template file: ", err)
+	}
+
+	currentHash := []byte(fmt.Sprintf("%d.%d", containerHash, templateHash))
+	filename := os.TempDir() + "/" + tempFile
+
+	previousHash, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.Println("unable to read tmp file: ", err)
+		log.Println("writing new file")
+	}
+
+	if string(previousHash) != string(currentHash) {
+		log.Println("changes detected")
+		if err := ioutil.WriteFile(filename, currentHash, 0666); err != nil {
+			log.Fatal(err)
+			cleanupTempFile()
```

Comment:
Do we actually need to crash here if we cannot write the file? 
Could just not write the file and continue to work.